### PR TITLE
Remove deprecations

### LINKED ISSUE #7530: Reminder to postpone deprecation of eager compilation of CUDA device functions.
As title. Context: https://github.com/numba/numba/pull/7496#discussion_r736484435

### PR DESCRIPTION
**Pull request by gmarkall**
This PR removes deprecation notices for things removed in 0.54 (and also removed things that were claimed to be removed in 0.54), and also updates the notes on items deprecated / removed in 0.55.

Fixes #7530, as it updates the relevant deprecation notice.

### CODE REVIEW

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
Fixes #7530, as it updates the deprecation notice.

File: docs/source/reference/deprecation.rst [hunk 1]
```diff
@@ -397,37 +288,26 @@ device function, then a device function should be compiled.
...
 Schedule
 --------
 
 - In Numba 0.54: Eager compilation of device functions will be deprecated.
 - In Numba 0.55: Eager compilation of device functions will be unsupported and
-  attempts to eagerly compile device functions will raise an error.
+  the provision of signatures for device functions will only enforce casting.
 
 
 .. _rocm_unmaintained:
```
Comment (stuartarchibald) [inline docs/source/reference/deprecation.rst:313]: Should this section also be removed, two releases have passed without comment?

Comment (gmarkall) [reply docs/source/reference/deprecation.rst:313]: Sure, I can remove this.

File: numba/core/config.py [hunk 2]
```diff
@@ -385,12 +385,6 @@ def avx_default():
         CUDA_PER_THREAD_DEFAULT_STREAM = _readenv(
             "NUMBA_CUDA_PER_THREAD_DEFAULT_STREAM", int, 0)
 
-        # Compute contiguity of device arrays using the relaxed strides
-        # checking algorithm.
-        NPY_RELAXED_STRIDES_CHECKING = _readenv(
-            "NUMBA_NPY_RELAXED_STRIDES_CHECKING",
-            int, 1)
-
         # HSA Configs
```
Comment (stuartarchibald) [inline numba/core/config.py:388]: Can probably remove this HSA config var as ROCm has been removed entirely now the deprecation notice has gone!

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the update, one minor last HSA/ROCm related removal to do if that's ok please? Else looks good.

File: numba/core/config.py (hunk 2 above)
Comment (gmarkall) [inline numba/core/config.py:388]:
```suggestion
```

Comment (stuartarchibald) [reply numba/core/config.py]: Many thanks.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes.

**Comment by esc** [comment]
`Build numba_smoketest_cuda_yaml_107 has started`

**Comment by esc** [comment]
Screen Shot 2021-12-03 at 10 12 04

All green :green

**Comment by stuartarchibald** [comment]
Thanks for testing the build @esc. Patch is good, marking for merge.

**Pull request merged**